code/game/objects/items/reagent_containers/hypospray.dm

```diff
@@ -56,12 +58,88 @@
+	update_icon()
+
+
+/obj/item/reagent_container/hypospray/advanced
+	icon_state = "hypo"
+	container_type = REFILLABLE|DRAINABLE
+	var/core_name = "hypospray"
+
+/obj/item/reagent_container/hypospray/advanced/tricordrazine
+	list_reagents = list("tricordrazine" = 60)
+
+/obj/item/reagent_container/hypospray/advanced/tricordrazine/New()
+	. = ..()
+	update_icon()
+
 
+/obj/item/reagent_container/hypospray/advanced/oxycodone
+	list_reagents = list("oxycodone" = 60)
+
+/obj/item/reagent_container/hypospray/advanced/oxycodone/New()
+	. = ..()
+	update_icon()
+
+/obj/item/reagent_container/hypospray/advanced/attack_self(mob/user)
+	var/str = copytext(reject_bad_text(input(user,"Label text?", "Set label", "")), 1, MAX_NAME_LEN)
+	if(!str || !length(str))
+		to_chat(user, "<span class='notice'>Invalid text.</span>")
+		return
+	to_chat(user, "<span class='notice'>You label [src] as \"[str]\".</span>")
+	name = "[core_name] ([str])"
+
+/obj/item/reagent_container/hypospray/advanced/update_icon()
+	overlays.Cut()
+
+	if(reagents.total_volume)
+		var/image/filling = image('icons/obj/reagentfillings.dmi', src, "[icon_state]10")
+
+		var/percent = round((reagents.total_volume / volume) * 100)
+		switch(percent)
+			if(0 to 9)			filling.icon_state = "[icon_state]-10"
+			if(10 to 24) 		filling.icon_state = "[icon_state]10"
+			if(25 to 49)		filling.icon_state = "[icon_state]25"
+			if(50 to 74)		filling.icon_state = "[icon_state]50"
+			if(75 to 79)		filling.icon_state = "[icon_state]75"
+			if(80 to 90)		filling.icon_state = "[icon_state]80"
+			if(91 to INFINITY)	filling.icon_state = "[icon_state]100"
+
+		filling.color = mix_color_from_reagents(reagents.reagent_list)
+		overlays += filling
+
+/obj/item/reagent_container/hypospray/advanced/examine(mob/user as mob)
+	if(get_dist(user,src) > 2)
+		to_chat(user, "\red You're too far away to see [src]'s reagent display!")
+		return
 
-/obj/item/reagent_container/hypospray/tricordrazine
-	desc = "The DeForest Medical Corporation hypospray is a sterile, air-needle autoinjector for rapid administration of drugs to patients. Contains tricordrazine."
-	volume = 30
-	list_reagents = list("tricordrazine" = 30)
\ No newline at end of file
+	if(!isnull(reagents))
+		var/dat = "\n \t \blue <b>Total Reagents:</b> [reagents.total_volume]/[volume]. <b>Dosage Size:</b> [min(reagents.total_volume, amount_per_transfer_from_this)]</br>"
+		if(reagents.reagent_list.len > 0)
+			for (var/datum/reagent/R in reagents.reagent_list)
+				var/percent = round(R.volume / max(0.01 , reagents.total_volume * 0.01),0.01)
+				var/dose = round(min(reagents.total_volume, amount_per_transfer_from_this) * percent * 0.01,0.01)
+				if(R.scannable)
+					dat += "\n \t \blue <b>[R.name]:</b> [R.volume]|[percent]% Amount per dose: [dose]</br>"
+				else
+					dat += "\n \t \blue <b>Unknown:</b> [R.volume]|[percent]% Amount per dose: [dose]</br>"
+		if(dat)
+			to_chat(user, "\blue [src]'s reagent display shows the following contents: [dat]")
```

Comment:
<span class = 'notice'> [...]</span> (blue), and remove the other istances of \blue.